### PR TITLE
MSET raw animation support

### PR DESCRIPTION
**Pull request by Xeeynamo**
A very basic implementation for animations that provides the minimum amount of necessary code to support animations. This is meant to help #256 .
animated model

PS. To simulate the same behaviour in the GIF, insert `objectId = 0x0555;` in ObjectEntity's constructor.

### CODE REVIEW

**Comment by kenjiuno** [comment]
Ok this pull request works well...

We can simulate `objectId = 0x0555;` by replacing `tt00.ard` with the one in [tt00.zip](https://github.com/Xeeynamo/OpenKh/files/5572773/tt00.zip), that is built with `OpenKh.Command.SpawnScript`

**Comment by Xeeynamo** [comment]
Thank you @kenjiuno , good idea about the ARD :) 

I converted this pull request to draft as there are some tweaks to the `Motion.cs` class and the way how I am animating the model. Currently I am parsing the 3D model at every frame, which is very inefficient. I want to ask: do we want to merge it anyway and tweak it with your #256 or is it better to squeeze more performance at the cost of making more code changes?

**Comment by kenjiuno** [comment]
> Currently I am parsing the 3D model at every frame, which is very inefficient. I want to ask: do we want to merge it anyway and tweak it with your #256 or is it better to squeeze more performance at the cost of making more code changes?

I agree with this idea.
This pull request contains fundamental and important code of skeletal animation support.
Currently this supports raw format (uncompressed?), but I believe that it will be easy to adapt with compressed format.

**Review by kenjiuno — APPROVED** [review]

File: OpenKh.Kh2/Motion.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        {
+            [Data] public int BoneCount { get; set; }
+            [Data] public int Unk14 { get; set; }
+            [Data] public int Unk18 { get; set; }
+            [Data] public int Unk1c { get; set; }
+            [Data] public int Unk20 { get; set; }
+            [Data] public int TotalFrameCount { get; set; }
+            [Data] public int Unk28 { get; set; }
+            [Data] public int Unk2c { get; set; }
+            [Data] public float Unk30 { get; set; }
```
Comment (kenjiuno) [inline OpenKh.Kh2/Motion.cs:29]: Unk30 ~ Unk4c seem to be a bounding box: min xyzw, and max xyzw

2020-11-21_09h56_46

Comment (Xeeynamo) [reply OpenKh.Kh2/Motion.cs:29]: Good info, thanks! Meanwhile I found Unk2C usage too: I will push the changes as soon as I turn on my computer.

**Comment by Xeeynamo** [comment]
Unk14, Unk18 and Unk1c are always 0. I was not able to find the meaning of Unk28 and [Unk2c is an offset to more matrices](https://github.com/Xeeynamo/OpenKh/pull/293/files#diff-7439ec36dcb3e053d943d4016864875882af62571bb86011d81bf174ef908fbfR115).

I want to write some documentation, but, I will wait for #297 to be merged and I will create a separate PR.

I wanted to do more performance improvements for this PR, but I did not want to create too much conflicts once we decide to rebase #256 . I am going to merge this :)

**Pull request merged**